packages/namekit-react/src/components/Link.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import cc from "classcat";
+
+export interface LinkProps extends React.ComponentPropsWithoutRef<"a"> {
+  className?: string;
+  variant?: "primary" | "secondary";
+  size?: "small" | "medium" | "large";
+  asChild?: React.ReactElement;
+}
+
+const linkBaseClasses = "nk-transition";
+
+const variantClasses = {
+  primary:
+    "nk-text-black nk-underline sm:nk-underline-offset-[4px] sm:nk-transition-all sm:nk-duration-200 sm:hover:nk-underline-offset-[2px]",
+  secondary: "nk-text-gray-500 hover:nk-text-black",
+};
+
+const sizeClasses = {
+  small: "nk-text-sm",
+  medium: "nk-text-base",
+  large: "nk-text-lg",
+};
+
+const isInternalLink = (href: string | undefined) => {
+  if (!href) return false;
```

Comment:
Is there any problem if a link is a "javascript link" rather than a "html link"?

For example, we can consider the "on click override" examples here: https://www.nameguard.io/docs/report

![CleanShot 2024-06-17 at 12 56 33](https://github.com/namehash/nameguard/assets/126201998/5db5427a-c2a7-4cd3-8831-30ba735b7132)

The above link is handled in Javascript rather than HTML.

Appreciate your advice on what we should do :+1: Also very open to the possibility that we need to make changes to how UI components such as `ReportBadge` and `ReportIcon` work :+1:

Please note that Franco is very close to merging a PR that makes a number of other changes to the UI components on https://www.nameguard.io/docs/report

* Link to PR: https://github.com/namehash/nameguard/pull/282
* Link to Preview Environment: https://nameguard-pnqz2xot4-namehash.vercel.app/docs/report

Reply:
I think this should be handled as a button or something else. No need to use the wrong semantic element.